vue-components/src/components/Table.vue

```diff
@@ -0,0 +1,187 @@
+<template>
+	<table
+		:class="[
+			'wikit',
+			'wikit-Table',
+			`wikit-Table--linear-${breakpoint}`
+		]"
+	>
+		<slot />
+	</table>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import { Breakpoint, validateBreakpoint } from './Breakpoint';
+
+export default Vue.extend( {
+	name: 'Table',
+	props: {
+		linearize: {
+			type: String as PropType<Breakpoint>,
+			validator( value: Breakpoint ): boolean {
+				return validateBreakpoint( value );
+			},
+			default: Breakpoint.Tablet,
+		},
+	},
+	computed: {
+		breakpoint(): string {
+			return validateBreakpoint( this.linearize ) ? this.linearize : 'tablet';
+		},
+	},
+} );
+</script>
+
+<style lang="scss">
+	@mixin linear-table {
+		/**
+		* Wipe the thead from the face of the earth
+		* modern screen readers will expose the
+		* generated content
+		*/
+		thead {
+			display: none;
+			visibility: hidden;
+		}
+
+		/**
+		* make everything display flex for alignment
+		*/
+		tbody,
+		tr,
+		th,
+		td {
+			height: auto;
+			display: flex;
+			flex-direction: column;
+		}
+
+		td,
+		th {
+			flex-direction: row;
+			flex-basis: 60%;
+		}
+
+		/**
+		* Labeling
+		*
+		* Adding a data-title attribute to the cells
+		* lets us add text before the content to provide
+		* the missing context.
+		*
+		* Markup:
+		*   <td data-title="Column Header">Content Here</td>
+		*
+		* Display:
+		*   Column Header: Content Here
+		*/
+		/* stylelint-disable selector-no-qualifying-type  */
```

Comment:
I think disabling this line from linting here is a good idea, since the motivation for this rule is usually to prevent more general style usages of qualifying types, that is - a tagname followed by a qualifier such as a class, id or other attribute selector.

However in this case we want to ensure that we are indeed only targeting `td`, and `th` with the `data-header` attribute (and not `trs` or any other element someone might put in that slot) so I think this usage is justified. Thank you!